artichoke-backend/src/extn/core/regexp/trampoline.rs

```diff
@@ -169,3 +172,23 @@ pub fn to_s(interp: &mut Artichoke, mut regexp: Value) -> Result<Value, Error> {
     let s = regexp.string();
     interp.try_convert_mut(s)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::test::prelude::*;
+
+    #[test]
+    fn should_raise_frozen_error() {
```

Comment:
I think this is the first test that has ever been added to a `trampoline` module. Thanks for blazing the trail!

We should add tests like these to the trampolines for `Array` and `String` too.

Reply:
Filed a ticket here:

- https://github.com/artichoke/artichoke/issues/2086

Reply:
Thank you for setting up pre-conditions for this PR as well as guiding me in the right direction.
It was a good opportunity to get a bit familiar with the internals of Artichoke.